src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,121 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final User EXISTING_USER = new User("BobAlison", "password", 25);
+    private static final User NEW_USER_WITH_EXISTING_LOGIN = new User(
+            "BobAlison", "password", 30);
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
+    private static final User NULL_USER = null;
+    private static final User VALID_USER = new User("NickAlison", "password", 65);
+    private static final User USER_WITH_NULL_AGE = new User("BobAlison", "password", null);
+    private static final User USER_WITH_NULL_PASSWORD = new User("BobAlison", null, 25);
+    private static final User USER_WITH_NULL_LOGIN = new User(null, "password", 25);
+    private static final User USER_WITH_SHORT_LOGIN = new User("abcde", "password", 25);
+    private static final User USER_WITH_SHORT_PASSWORD = new User("BobAlison", "abcde", 25);
+    private static final User USER_BELOW_MIN_AGE = new User("BobAlison", "password", 17);
+    private static final User USER_WITH_NEGATIVE_AGE = new User("NickAlison", "password", -16);
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    public void setup() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void testRegister_ValidUser_returnsRegisteredUser() {
```

Comment:
be careful with method/test names 
Word that follows underscore should start from lower case. Fix all of it 
```suggestion
    public void testRegister_validUser_returnsRegisteredUser() {
```